hw/drivers/sensors/bmp388/src/bmp388.c

```diff
@@ -3366,6 +3366,215 @@ bmp388_stream_read(struct sensor *sensor,
     return rc;
 }
 
+int
+bmp388_hybrid_read(struct sensor *sensor,
+                   sensor_type_t sensor_type,
+                   sensor_data_func_t read_func,
+                   void *read_arg,
+                   uint32_t time_ms)
+{
+    int rc;
+    struct bmp388 *bmp388;
+    struct bmp388_cfg *cfg;
+    os_time_t time_ticks;
+    os_time_t stop_ticks = 0;
+    uint16_t try_count;
+    
+#if MYNEWT_VAL(BMP388_FIFO_ENABLE)
+    /* FIFO object to be assigned to device structure */
+    struct bmp3_fifo fifo;
+    /* Pressure and temperature array of structures with maximum frame size */
+    struct bmp3_data sensor_data[74];
```

Comment:
This frame size should be a syscfg to make it configurable at build time.

Reply:
Also, there is possibility of a buffer overrun in this case mainly because the FIFO is of 512 bytes. If you keep reading it and the entire FIFO is filled up, there is going to be a FIFO overrun. Please try to limit your reads based on the number of data samples in the buffer as well.

Reply:
As discussed out of band, this is an array of bmp3_data structs.  I did move the magic number to a sysconfig, but I left it as 74.  The DS indicates the maximum frame length is 9 bytes. A data frame is 7 bytes (4 pressure, 2 temp, 1 header) and the sensor_time frame is 4 (3 time, 1 header).  I don't have insight into the historical reason for using 74 as a magic number, but if we did I suspect the correct number is 85 as that would be 510 bytes.

Reply:
sensor_time is not part of the FIFO frames, as per the BMP388 Datasheet "The data for the sensor-time frame consists of register sensor_time content at the time the sensortime frame transmission has started. A sensor-time frame is not stored in the FIFO, but append to every FIFO burst read operation after all data has been transmitted if fifo_time_en=‘1’."

Reply:
Per out of band discussion this will stay as is.